[Xcode] Teach "Create symlinks to XPC services" build phase to also symlink libWebKitSwift.dylib

aestes opened the pull request:
#### 58b61c2b391439afb6e58cf7fc9e20d5a0e82b3a
<pre>
[Xcode] Teach &quot;Create symlinks to XPC services&quot; build phase to also symlink libWebKitSwift.dylib
<a href="https://bugs.webkit.org/show_bug.cgi?id=269229">https://bugs.webkit.org/show_bug.cgi?id=269229</a>
<a href="https://rdar.apple.com/122825232">rdar://122825232</a>

Reviewed by Elliott Williams.

libWebKitSwift.dylib expects to be installed in WebKit.framework/Frameworks, but nothing in the
engineering build system places it in that location. While WebKitSwiftSoftLink.mm will find the
dylib in $BUILT_PRODUCTS_DIR on macOS, tools for creating roots from engineering built products
(e.g., `package-root`) do not know to include the dylib without at least a symlink to it in the
expected location. To fix this, extended WebKit&apos;s &quot;Create symlinks to XPC services&quot; build phase
to create a symlink to libWebKitSwift.dylib in the expected install location.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/274514@main">https://commits.webkit.org/274514@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e939c7bf0790fb715710e324762054c0f11a6551

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39281 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18260 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41634 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41815 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/35181 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21118 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15589 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32860 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39855 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15368 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34046 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13355 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13321 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43093 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35663 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35315 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39138 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14093 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11629 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37378 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15699 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8793 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/15382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15188 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18420,10 +18420,11 @@
 				"$(BUILT_PRODUCTS_DIR)/$(XPCSERVICES_FOLDER_PATH)/com.apple.WebKit.Networking.xpc",
 				"$(BUILT_PRODUCTS_DIR)/$(XPCSERVICES_FOLDER_PATH)/com.apple.WebKit.GPU.xpc",
 				"$(BUILT_PRODUCTS_DIR)/$(XPCSERVICES_FOLDER_PATH)/com.apple.WebKit.Model.xpc",
+				"$(BUILT_PRODUCTS_DIR)/WebKit.framework/Frameworks/libWebKitSwift.dylib",
```
On macOS, this path will be `$(BUILT_PRODUCTS_DIR)/WebKit.framework/Versions/A/Frameworks/libWebKitSwift.dylib`, and the build system doesn't look through symlinks (since it is responsible for creating the symlink).

You probably want to add a `$(WK_FRAMEWORK_VERSION_PREFIX)` to the string, or one of the existing path macros like `$(WEBKIT_FRAMEWORKS_DIR)`, although I think the latter only works in production builds.

emw-apple reviewed (commented):
> libWebKitSwift.dylib expects to be installed in WebKit.framework/Frameworks

Is it always a top-level `Frameworks/` directory in the bundle, or does it follow the normal macOS convention of being in a Versions/A/ directory? If so, you probably want to take my suggestion below:

emw-apple reviewed (approved):
r+ with fix mentioned above

aestes committed:
[Xcode] Teach "Create symlinks to XPC services" build phase to also symlink libWebKitSwift.dylib
https://bugs.webkit.org/show_bug.cgi?id=269229
rdar://122825232

Reviewed by Elliott Williams.

libWebKitSwift.dylib expects to be installed in WebKit.framework/Frameworks, but nothing in the
engineering build system places it in that location. While WebKitSwiftSoftLink.mm will find the
dylib in $BUILT_PRODUCTS_DIR on macOS, tools for creating roots from engineering built products
(e.g., `package-root`) do not know to include the dylib without at least a symlink to it in the
expected location. To fix this, extended WebKit's "Create symlinks to XPC services" build phase
to create a symlink to libWebKitSwift.dylib in the expected install location.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/274514@main
